Fix handleTagChange

noahmalmed opened the pull request:
Staging branch URL: https://fix_4801.pfe-preview.zooniverse.org/

Fixes #4801 Fixes #4907

Just moved this to be an inline function, so the tag array doesn't get converted to a string. @srallen is this the type of thing you had in mind as a solution?

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

eatyourgreens commented:
This is a duplicate of #4799, which bypassed the bug by bypassing DOMPurify. It's dangerous to pass unsanitised user input to Panoptes. I'd recommend sanitising the tags by passing something like `{tags: value.map(tag => DOMPurify.sanitise(tag))}` to the update method.

https://github.com/zooniverse/Panoptes-Front-End/pull/4799#issuecomment-414983273

noahmalmed commented:
👍 that makes sense, I'll update this

noahmalmed commented:
-Updated to sanitize each member of the tag array
- Updated to sanitize the array before we update state, so the sanitized values are displayed to the user

eatyourgreens commented on app/lib/sanitize-array-input.coffee:
```diff
@@ -0,0 +1,7 @@
+createDOMPurify = require('dompurify');
```
Can this helper be written in JS? We're dropping coffeescript in favour of ES6.

eatyourgreens commented on app/lib/sanitize-array-input.coffee:
```diff
@@ -0,0 +1,7 @@
+
+module.exports = (array) ->
+  sanitizedArray = array.map (string) -> DOMPurify.sanitize(string)
+  return sanitizedArray.filter (string) -> string != ''
```
Use strict inequality (!==) here to clean up empty strings.

noahmalmed commented:
@eatyourgreens changes made per your request
